src/common/reducers/entities.js

```diff
@@ -0,0 +1,66 @@
+import merge from 'lodash/merge';
+
+import * as ActionTypes from '../actions/repository';
+
+// export only for testing
+export function repository(state={
+  __isSelected: false,
+  __isFetching: false,
+  __error: null
+}, action) {
+
+  switch(action.type) {
+    case ActionTypes.REPOSITORY_TOGGLE_SELECT: {
+      const wasSelected = state.__isSelected;
+
+      return {
+        ...state,
+        __isSelected: !wasSelected
+      };
+    }
+    case ActionTypes.REPOSITORY_BUILD: {
+      return {
+        ...state,
+        __isFetching: true,
+        __isSelected: true
+      };
+    }
+    case ActionTypes.REPOSITORY_SUCCESS:
```

Comment:
`REPOSITORY_SUCCESS` and `REPOSITORY_FAILURE` are a bit vague. What succeeded and failed?

Can I suggest changing this to:
`REPOSITORY_BUILD_REQUEST`, `REPOSITORY_BUILD_SUCCESS`, and `REPOSITORY_BUILD_FAILURE`?